.github/workflows/main.yml

```diff
@@ -6,41 +6,83 @@ on:
       - '*'
     tags:
       - '*'
+
 jobs:
-  build:
-    name: Build
+  snapshot:
+    name: Snapshot build
     runs-on: ubuntu-latest
+    if: ${{ ! startsWith(github.ref, 'refs/tags/') }}
     steps:
+      # Checkout so the workflow can access the repository
       - name: Checkout
-        uses: actions/checkout@v2
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 #v4.1.1
+      # Setup up Go utilities
       - name: Set up Go
-        uses: actions/setup-go@v2
+        uses: actions/setup-go@0c52d547c9bc32b1aa3301fd7a9cb496313a4491 #v5.0.0
```

Comment:
What's the reasoning behind replacing all the tags with hashes?

Reply:
Referencing an action with commit hashes is more secure than referencing with tags.
If a malicious actor is controlling the repo of the action at some point, a tag content could change, a commit content couldn't.

See this blog article on the subject for more details : https://blog.rafaelgss.dev/why-you-should-pin-actions-by-commit-hash

The only exception in the changes we did on the workflows is the SLSA generator which specify in it's documentation that it must be referenced by a tag.